docs/FIELDS.md

```diff
@@ -0,0 +1,338 @@
+# List of fields
+
+## Generators
+
+### `configMapGenerator`
+
+Each entry in this list results in the creation of
+one ConfigMap resource (it's a generator of n maps).
+The example below creates two ConfigMaps. One with the
+names and contents of the given files, the other with
+key/value as data.
+
+Usage:
+
+```yaml
+configMapGenerator:
+- name: myJavaServerEnvVars
+  literals:	
+  - JAVA_HOME=/opt/java/jdk
+  - JAVA_TOOL_OPTIONS=-agentlib:hprof
+- name: myJavaServerProps
+  files:
+  - application.properties
+  - more.properties
+```
+
+### `secretGenerator`
+
+Each entry in this list results in the creation of
+one Secret resource (it's a generator of n secrets).
+A command can do anything to get a secret,
+e.g. prompt the user directly, start a webserver to
+initate an oauth dance, etc.
+
+Usage:
+
+```yaml
+secretGenerator:
+- name: app-tls
+  commands:
+    tls.crt: "cat secret/tls.cert"
+    tls.key: "cat secret/tls.key"
+  type: "kubernetes.io/tls"
+- name: downloaded_secret
+  # timeoutSeconds specifies the number of seconds to
+  # wait for the commands below. It defaults to 5 seconds.
+  timeoutSeconds: 30
+  commands:
+    username: "curl -s https://path/to/secrets/username.yaml"
+    password: "curl -s https://path/to/secrets/password.yaml"
+  type: Opaque
+- name: env_file_secret
+  # envCommand is similar to command but outputs lines of key=val pairs
+  # i.e. a Docker .env file or a .ini file.
+  # you can only specify one envCommand per secret.
+  envCommand: printf "DB_USERNAME=admin\nDB_PASSWORD=somepw"
+  type: Opaque
+```
+
+## Input resources
```

Comment:
This `Input resources` is confusing as you listed `resources`, `bases` and `crds` under it.

Reply:
agree, a better word: _operands_